app/interactors/file_attachments/update_interactor.rb

```diff
@@ -0,0 +1,63 @@
+class FileAttachments::UpdateInteractor < ApplicationInteractor
+  delegate :params,
+           :user,
+           :edition,
+           :file_attachment_revision,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      find_file_attachment
+
+      update_file_attachment
+      update_edition
+
+      create_timeline_entry
+      update_preview
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+    assert_edition_state(edition, &:editable?)
+  end
+
+  def find_file_attachment
+    context.file_attachment_revision = edition.file_attachment_revisions
+                                              .find_by!(file_attachment_id: params[:file_attachment_id])
+  end
+
+  def update_file_attachment
+    updater = Versioning::FileAttachmentRevisionUpdater.new(file_attachment_revision, user)
+    revision_attributes = attachment_params.slice(:unique_reference)
+    updater.assign(revision_attributes)
+
+    context.file_attachment_revision = updater.next_revision
+  end
+
+  def update_edition
+    updater = Versioning::RevisionUpdater.new(edition.revision, user)
+    updater.update_file_attachment(file_attachment_revision)
+
+    context.fail!(unchanged: true) unless updater.changed?
+
+    EditDraftEditionService.call(edition, user, revision: updater.next_revision)
+    edition.save!
+  end
+
+  def create_timeline_entry
+    TimelineEntry.create_for_revision(entry_type: :file_attachment_updated,
```

Comment:
Turns out this is actually consistent with [the way we do timeline entries for images](https://github.com/alphagov/content-publisher/blob/master/app/interactors/images/update_crop_interactor.rb#L50). I suppose this makes sense because all this stuff is about updating the attachment, and we could have  it all on one form (as a counter-argument to having different types).